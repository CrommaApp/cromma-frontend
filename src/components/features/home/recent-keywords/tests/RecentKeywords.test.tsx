import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RecentKeywords from '@components/features/home/recent-keywords';
import userEvent from '@testing-library/user-event';

describe('RecentKeywords', () => {
	const moveToReultPage = jest.fn();

	const mockRecentKeywords: string[] = ['korea', 'seoul'];

	beforeEach(() => {
		render(<RecentKeywords recentKeywords={mockRecentKeywords} moveToReultPage={moveToReultPage} />);
	});

	it('Snapshot', () => {
		const component = renderer.create(
			<RecentKeywords recentKeywords={mockRecentKeywords} moveToReultPage={moveToReultPage} />,
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Show recent keyword list delivered through props', () => {
		const keywords = screen.getAllByRole('listitem');

		expect(keywords).toHaveLength(2);

		expect(keywords[0]).toHaveTextContent('korea');

		expect(keywords[1]).toHaveTextContent('seoul');
	});

	it('Move to result page when click keyword', () => {
		const keywords = screen.getAllByRole('listitem');

		userEvent.click(keywords[0]);

		expect(moveToReultPage).toHaveBeenCalledWith('korea');
	});
});
