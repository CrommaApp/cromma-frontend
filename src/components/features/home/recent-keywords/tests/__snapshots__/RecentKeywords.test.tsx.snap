// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentKeywords Snapshot 1`] = `
<ul
  className="sc-bdvvtL ktEXkV"
>
  <li
    onClick={[Function]}
  >
    korea
  </li>
  <li
    onClick={[Function]}
  >
    seoul
  </li>
</ul>
`;
